### ISSUE TITLE
http resource not working

### ISSUE THREAD

**Issue by g1ps** [inspec/inspec#3608]
### 🎛 Description

🐛 bug report

Expected behaviour: It should work.
Actual behaviour: It doesn't

### 🌍 InSpec and Platform Version

Inspec 3.0.12
RHEL 7

### 🤔 Replication Case
This trivial case doesn't work:
````
describe http('http://www.example.com') do
  its('status') { should cmp 200 }
end
````

### 💻 Stacktrace
Output:
````
Traceback (most recent call last):
        26: from /usr/bin/inspec:296:in `<main>'
        25: from /usr/bin/inspec:296:in `load'
        24: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.0.12/bin/inspec:12:in `<top
(required)>'
        23: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `
start'
        22: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispa
tch'
        21: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/invocation.rb:12
6:in `invoke_command'
        20: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/command.rb:27:in
 `run'
        19: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.0.12/lib/inspec/cli.rb:184:i
n `exec'
        18: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.0.12/lib/inspec/runner.rb:11
0:in `run'
        17: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.0.12/lib/inspec/runner.rb:10
3:in `load'
        16: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.0.12/lib/inspec/runner.rb:10
3:in `each'
        15: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.0.12/lib/inspec/runner.rb:10
4:in `block in load'
        14: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.0.12/lib/inspec/runner.rb:28
3:in `register_rule'
        13: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.0.12/lib/inspec/runner.rb:28
3:in `flat_map'
        12: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.0.12/lib/inspec/runner.rb:28
3:in `each'
        11: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.0.12/lib/inspec/runner.rb:28
4:in `block in register_rule'
        10: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.0.12/lib/inspec/runner.rb:27
2:in `get_check_example'
         9: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.0.12/lib/inspec/runner.rb:31
4:in `rspec_failed_block'
         8: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.0.12/lib/inspec/runner_rspec
.rb:27:in `example_group'
         7: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/rspec-core-3.8.0/lib/rspec/core/examp
le_group.rb:260:in `block in define_example_group_method'
         6: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/rspec-core-3.8.0/lib/rspec/core/examp
le_group.rb:386:in `subclass'
         5: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/rspec-core-3.8.0/lib/rspec/core/examp
le_group.rb:422:in `set_it_up'
         4: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/rspec-core-3.8.0/lib/rspec/core/metad
ata.rb:258:in `create'
         3: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/rspec-core-3.8.0/lib/rspec/core/metad
ata.rb:133:in `populate'
         2: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/rspec-core-3.8.0/lib/rspec/core/metad
ata.rb:180:in `build_description_from'
         1: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.0.12/lib/resources/http.rb:6
6:in `to_s'
/opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.0.12/lib/resources/http.rb:62:in `http_method
': undefined method `fetch' for nil:NilClass (NoMethodError)
````

**Comment by jquick** [comment]
Seems to be an issue with `faraday` on your machine. I will see if I can reproduce on my end.

**Comment by g1ps** [comment]
Thanks. I don't even know what that is. I haven't installed it myself so if it's not in the package it's probably missing.

**Comment by jquick** [comment]
Yeah its just a gem that comes with the inspec bundle so it should be provided for you. If your running remote (via `-t ssh://...`) it would use CURL but I think you are running locally here.

**Comment by g1ps** [comment]
I installed 3.0.52 on a clean RHEL VM. Same result. Not good. Is there an older version I can use which will work?

**Comment by g1ps** [comment]
And no, not running locally, although I probably tried that. I tried to cover all the bases. Thanks.

**Comment by jquick** [comment]
I am having issues reproducing this on a fresh 7.5 VM:
## Test file
```
jquick@Jareds-MBP:~/Chef/inspec/_test
(master)> cat http_test.rb
describe http('http://www.google.com') do
  its('headers.Content-Type') { should cmp 'text/html; charset=ISO-8859-1' }
end

describe http('http://www.example.com') do
  its('status') { should cmp 200 }
end

describe os do
  its('name') { should eq 'redhat' }
  its('release') { should eq '7.5' }
end
```

## Run
```
jquick@Jareds-MBP:~/Chef/inspec/_test
(master)> inspec exec http_test.rb -t ssh://vagrant:vagrant@127.0.0.1:2200

Profile: tests from http_test.rb (tests from http_test.rb)
Version: (not specified)
Target:  ssh://vagrant@127.0.0.1:2200

  http GET on http://www.google.com
     ✔  headers.Content-Type should cmp == "text/html; charset=ISO-8859-1"
  http GET on http://www.example.com
     ✔  status should cmp == 200
  Operating System Detection
     ✔  name should eq "redhat"
     ✔  release should eq "7.5"

Test Summary: 4 successful, 0 failures, 0 skipped
```
Using https://app.vagrantup.com/generic/boxes/rhel7 as my image.

Still digging around for potential issues.

**Comment by g1ps** [comment]
I tried reverting to Inspec 2.3.24. This doesn't work there, either:
````
C:/opscode/inspec/embedded/lib/ruby/gems/2.5.0/gems/inspec-2.3.24/lib/resources/http.rb:62:in `http_method': undefined method `fetch' for nil:NilClass (NoMethodError)
````

**Comment by clintoncwolfe** [comment]
Could be #2822, in which the resource tries to use curl on the remote machine?

**Comment by g1ps** [comment]
I tried this again today on a RHEL master VM running 3.0.12 and it worked. This is the same machine I used previously. I have no idea what changed as I've juggled so many things chasing Inspec issues that I've lost track. I may have reverted and/or updated the version on this machine. I am running the test locally on the master. if I tell it run on an intermediate Windows machine with `-t`, it fails as above.
